bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/internal/findandreplace/FindReplaceLogic.java

```diff
@@ -169,7 +169,7 @@ private void useSelectedLines(boolean selectedLines) {
 		if (selectedLines) {
 
 			IRegion scope;
-			if (oldScope == null) {
+			if (target.getSelection().y > 1 || oldScope == null) {
 				Point lineSelection = extensionTarget.getLineSelection();
```

Comment:
What is the reason for not using the currently selected line if there is no selection but just the cursor being in some line? I would expect that the line in which the cursor is currently placed is taken as selected line when activating the option.

As a more general question: why do we need to store the `oldScope` at all? I would expect scope to be set once the option is activated and whenever it is reset to then use the selection at that point in time (rather than an old one).